File: CSMApps/Vertebrae/v1.0.0/src/main/java/com/carrpod/vertebrae/VertebraeApplication.java
package com.carrpod.vertebrae;

import android.app.Application;
import android.content.Context;
import android.content.Intent;
import android.os.Build;

import com.carrpod.vertebrae.service.SessionManagerService;
import com.carrpod.vertebrae.service.HeartbeatService;
import com.carrpod.vertebrae.service.FloatingWindowService;
import com.carrpod.vertebrae.comm.InterSessionServer;
import com.carrpod.vertebrae.storage.SessionStorageManager;
import com.carrpod.vertebrae.network.WebSocketManager;
import com.carrpod.vertebrae.comm.SessionCommunicator;

public class VertebraeApplication extends Application {

    private static VertebraeApplication instance;

    public static VertebraeApplication getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;

        SessionStorageManager.initialize(this);
        WebSocketManager.initialize(this);
        SessionCommunicator.initialize(this);
        
        // Services started from SplashActivity after permissions
    }
}

File: CSMApps/Vertebrae/v1.0.0/src/main/java/com/carrpod/vertebrae/VertebraeApplication.kt
package com.carrpod.vertebrae

import android.app.Application
import android.content.Context
import com.carrpod.vertebrae.service.VertebraeCoreService
import com.carrpod.vertebrae.storage.SessionStorage
import com.carrpod.vertebrae.network.WebSocketManager
import com.carrpod.vertebrae.comm.SessionCommunicator

class VertebraeApplication : Application() {

    companion object {
        lateinit var instance: VertebraeApplication
            private set
    }

    override fun onCreate() {
        super.onCreate()
        instance = this

        SessionStorage.initialize(this)
        WebSocketManager.initialize(this)
        SessionCommunicator.initialize(this)

        startCoreServices()
    }

    private fun startCoreServices() {
        val intent = android.content.Intent(this, VertebraeCoreService::class.java)
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
            startForegroundService(intent)
        } else {
            startService(intent)
        }
    }
}